### PR TITLE
Make bswap constexpr

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: include/intx/int128.hpp [hunk 1]
```diff
@@ -542,16 +542,22 @@ inline constexpr unsigned clz(uint128 x) noexcept
 }
 
 
-inline uint64_t bswap(uint64_t x) noexcept
+inline constexpr uint64_t bswap(uint64_t x) noexcept
 {
-#ifdef _MSC_VER
-    return _byteswap_uint64(x);
-#else
+#if __has_builtin(__builtin_bswap64)
```
Comment (axic) [inline include/intx/int128.hpp:547]: Which compilers support this?

Comment (chfast) [reply include/intx/int128.hpp:547]: GCC and clang

Comment (axic) [reply include/intx/int128.hpp:547]: I mean versions :wink:

Comment (chfast) [reply include/intx/int128.hpp:547]: You mean `__has_builtin` or `__builtin_bswap64`?

Comment (axic) [reply include/intx/int128.hpp:547]: Both. On what compilers will that codepath be enabled? As that is our preferred codepath I reckon.

**Comment by chfast** [comment]
The `__has_builtin` is supported in Clang 4+ and GCC 10+.
GCC 6+ will not use the `__builtin_bswap` then, be it can optimize the generic implementation to `bswap` instruction.
The intrinsic in MSVC works as expected.
https://godbolt.org/z/W56eWKE9r

**Review by axic — APPROVED** [review]

**Pull request merged**